external/kafka/src/test/scala/org/apache/spark/streaming/kafka/KafkaStreamSuite.scala

```diff
@@ -65,12 +67,14 @@ class KafkaStreamSuite extends SparkFunSuite with Eventually with BeforeAndAfter
 
     val stream = KafkaUtils.createStream[String, String, StringDecoder, StringDecoder](
       ssc, kafkaParams, Map(topic -> 1), StorageLevel.MEMORY_ONLY)
-    val result = new mutable.HashMap[String, Long]() with mutable.SynchronizedMap[String, Long]
+    val result = new ConcurrentHashMap[String, Long].asScala
     stream.map(_._2).countByValue().foreachRDD { r =>
       val ret = r.collect()
       ret.toMap.foreach { kv =>
-        val count = result.getOrElseUpdate(kv._1, 0) + kv._2
-        result.put(kv._1, count)
+        result.synchronized {
+          val count = result.getOrElseUpdate(kv._1, 0) + kv._2
```

Comment:
@holdenk 
Thanks for your quick reply. 
I initially changed 
   val count = result.getOrElseUpdate(kv._1, 0) + kv._2
to
    result.putIfAbsent(kv._1, 0)
    val count = result.get(kv._1) + kv._2_
but the test failed for me. I guess a different thread can come in between of the two lines and the concurrency is not guaranteed any more.  So I used synchronized block instead.